web/modules/custom/weather_block/src/Plugin/Block/WeatherBlock.php

```diff
@@ -151,57 +163,29 @@ public function blockForm($form, FormStateInterface $form_state) {
 
   /**
    * {@inheritdoc}
+   *
+   * @throws \Exception
    */
   public function blockSubmit($form, FormStateInterface $form_state): void {
+    $user_id = $this->accountProxy->id();
+    $cities = array_map('trim', explode(',', $form_state->getValue([
+      'city_selection',
+      'cities',
+    ])));
+    $this->cityService->saveCitiesForUser($user_id, $cities);
```

Comment:
Again, cities, buy you showed me that the user was able to select only one city.